core/amber/src/main/python/core/architecture/packaging/output_manager.py

```diff
@@ -56,13 +62,28 @@ def __init__(self, worker_id: str):
         }
         self._ports: typing.Dict[PortIdentity, WorkerPort] = dict()
         self._channels: typing.Dict[ChannelIdentity, Channel] = dict()
+        self._port_result_writers: typing.Dict[PortIdentity, PortResultWriter] = dict()
 
-    def add_output_port(self, port_id: PortIdentity, schema: Schema) -> None:
+    def add_output_port(
+        self, port_id: PortIdentity, schema: Schema, storage_uri: str
+    ) -> None:
         if port_id.id is None:
             port_id.id = 0
         if port_id.internal is None:
             port_id.internal = False
 
+        if storage_uri != "":
+            document: VirtualDocument[Tuple]
+            document, _ = DocumentFactory.open_document(storage_uri)
+            writer = document.writer(str(get_worker_index(self.worker_id)))
+            writer_thread = PortResultWriter(writer)
```

Comment:
this is not a thread. this is a runnable, right? I see you start a thread in line 80. I think better to name it properly.

Reply:
Renamed